### PR TITLE
Add spring/decompress module

### LINKED ISSUE #3906: new subworkflow: spring/decompress
### Is there an existing subworkflow for this?

- [X] I have searched for the existing subworkflow

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this subworkflow, I added myself to the `Assignees` to facilitate tracking who is working on the subworkflow

### PR DESCRIPTION
**Pull request by xec-cm**
## PR checklist

Closes #3906

- [X] This comment contains a description of changes (with reason).
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [X] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/nf-core/spring/decompress/test.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+- name: spring decompress test_spring_decompress_single_end
+  command: nextflow run ./tests/modules/nf-core/spring/decompress -entry test_spring_decompress_single_end -c ./tests/config/nextflow.config
+  tags:
+    - spring
+    - spring/decompress
+  files:
+    - path: output/spring/test.fastq.gz
+      md5sum: e71acb18cb172037d021daf2801e6d18
+    - path: output/spring/test.spring
+    - path: output/spring/versions.yml
```
Comment (klkeys) [inline tests/modules/nf-core/spring/decompress/test.yml:10]: odd that this file would not have a stable MD5 sum, can you confirm?

**Review by klkeys — CHANGES_REQUESTED** [review]
can you confirm the comments about MD5 stability of your `versions.yml` files?

other than that, the PR looks pretty clean 👌🏼

File: tests/modules/nf-core/spring/decompress/test.yml (hunk 1 above)
Comment (SPPearce) [reply tests/modules/nf-core/spring/decompress/test.yml:10]: The `nf-core tools modules create-test-yml` command by default does not include the md5sum for the versions.yml file. This means that bumping the version of the tool doesn't require updating the md5sums unless something has changed in the output.

Comment (klkeys) [reply tests/modules/nf-core/spring/decompress/test.yml:10]: seems reasonable, going to resolve comment then

**Review by klkeys — APPROVED** [review]

**Comment by xec-cm** [comment]
thanks @klkeys and @SPPearce for the review!

**Pull request merged**